HDFDataset.py

```diff
@@ -1039,15 +1047,18 @@ def dump_from_dataset(self, dataset, epoch=1, start_seq=0, end_seq=float("inf"),
         progress_bar_with_time(float(i) / num_seqs)
 
     print("Set seq len info...", file=log.v3)
-    hdf_dataset.create_dataset(attr_seqLengths, shape=(num_seqs, 2), dtype="int32")
+    hdf_dataset.create_dataset(attr_seqLengths, shape=(num_seqs, 1+len(data_target_keys)), dtype="int32")
     for i, seq_len in enumerate(seq_lens):
       data_len = seq_len[default_data_input_key]
-      targets_len = seq_len[default_data_target_key]
-      for data_key in data_target_keys:
-        assert seq_len[data_key] == targets_len, "different lengths in multi-target not supported"
-      if targets_len is None:
-        targets_len = data_len
-      hdf_dataset[attr_seqLengths][i] = [data_len, targets_len]
+      if len(data_target_keys) > 0:
+        targets_len = seq_len[default_data_target_key]
+        for data_key in data_target_keys:
+          assert seq_len[data_key] == targets_len, "different lengths in multi-target not supported"
```

Comment:
This is actually not needed anymore with your change of the `attr_seqLengths` shape above.

Reply:
Which change exactly are you talking about? Is this still relevant in the updated commit?

Reply:
The check that all the seq lens are the same. This was needed for the old shape of seq lengths.

Reply:
Ping?

Reply:
I'm sorry, I totally forgot about this. 
I don't have any time to do this properly right, but hopefully after I handed in my thesis.

Reply:
Why is that `assert` still needed? It's actually wrong, isn't it?